src/python_testing/TC_OPCREDS_3_5.py

```diff
@@ -0,0 +1,246 @@
+                         "Verify that DUT sends ArmFailSafeResponse with the ErrorCode set to OK"),
+                TestStep(12, "TH1 Sends CSRRequest command with the IsForUpdateNOC field set to true",
+                         "Verify that the DUT returns a CSRResponse and save as csr_update2"),
+                TestStep(13, """TH1 generates a new NOC and ICAC with the following properties:
+
+                                - new NOC is generated from the NOCSR returned in csr_update2. The subject is set to match noc_original. The NOC is signed by new ICA. Save as noc_update2
+
+                                - new ICAC must be distinguishable from icac_original and icac_update1. This can be accomplished by changing any of the following: the subject, subject public key or validity. The certificate must be valid. ICAC is signed by the original key for trusted_root_original. Save as icac_update2"""),
+                TestStep(14, """TH1 sends the UpdateNOC command to the Node Operational Credentials cluster with the following fields:
+
+                                - NOCValue is set to noc_update2
+
+                                - ICACValue is to set icac_update2""", "Verify that the DUT responds with a NOCResponse having its StatusCode field set to Ok"),
+                TestStep(15, "TH1 reads the NOCs attribute from the Node Operational Credentials cluster using a fabric-filtered read",
+                         "Verify that the returned list has a single entry. Verify that the NOC field matches noc_update2 and the ICAC field matches icac_update2"),
+                TestStep(16, "TH1 sends the CommissioningComplete command to the General Commissioning cluster",
+                         "Verify that the DUT returns a CommissioningCompleteResponse with the ErrorCode set to OK"),
+                TestStep(17, "TH1 sends ArmFailSafe command to the DUT with the ExpiryLengthSeconds field set to 0 to verify that the CommissioningComplete command successfully persisted the failsafe context.",
+                         "Verify that DUT sends ArmFailSafeResponse with the ErrorCode set to OK"),
+                TestStep(18, "TH1 reads the NOCs attribute from the Node Operational Credentials cluster using a fabric-filtered read",
+                         "Verify that the returned list has a single entry. Verify that the NOC field matches noc_update2 and the ICAC field matches icac_update2"),
+                TestStep(19, "TH1 reads its fabric index from the CurrentFabricIndex attribute and saves as fabric_idx"),
+                TestStep(20, "TH0 sends the RemoveFabric command with the fabric index set to fabric_idx")
+                ]
+
+    @async_test_body
+    async def test_TC_OPCREDS_3_5(self):
+        opcreds = Clusters.OperationalCredentials
+
+        self.step(1)
+
+        # Start with 10 years for cert validity
+        original_cert_validity = 365*24*60*60*10
```

Comment:
this is repeated in many places now. Would it make sense to have a constant?

Reply:
Actually, the test needs to modify this value three times, using 10 years for the first certificate, 9 for the next, and 8 for the last. Because of this, I think it’s better not to use a constant in the test. But, I have added it as a constant in the `CertificateAuthority` class since it’s used in many places there.

I have also replaced the calculations with `timedelta` as you suggested. Let me know if you have any concerns.